pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -17,359 +17,172 @@
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// StoragePoolVolumeProtectionFinalizer makes sure that no volume is present in the pool at the time of storage
+	// pool CR deletion.
+	StoragePoolVolumeProtectionFinalizer = "openebs.io/volume-protection"
+
+	// StoragePoolSourceProtectionFinalizer makes sure that internal pool resource is not present at the time of
+	// storage pool CR deletion.
+	StoragePoolSourceProtectionFinalizer = "openebs.io/pool-source-protection"
```

Comment:
Is it better (recommended) to have a single deletion finalizer that the reconciler (watcher) will delete after all necessary checks and actions are performed? What's the standard way?